sdk/appconfiguration/app-configuration/src/generated/src/appConfiguration.ts

```diff
@@ -1,764 +1,816 @@
 /*
- * Copyright (c) Microsoft Corporation. All rights reserved.
- * Licensed under the MIT License. See License.txt in the project root for
- * license information.
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
  *
  * Code generated by Microsoft (R) AutoRest Code Generator.
- * Changes may cause incorrect behavior and will be lost if the code is
- * regenerated.
+ * Changes may cause incorrect behavior and will be lost if the code is regenerated.
  */
 
 import * as coreHttp from "@azure/core-http";
-import * as Models from "./models";
-import * as Mappers from "./models/mappers";
 import * as Parameters from "./models/parameters";
+import * as Mappers from "./models/mappers";
 import { AppConfigurationContext } from "./appConfigurationContext";
+import {
+  AppConfigurationOptionalParams,
+  ApiVersion10,
+  AppConfigurationGetKeysOptionalParams,
+  AppConfigurationGetKeysResponse,
+  AppConfigurationCheckKeysOptionalParams,
+  AppConfigurationCheckKeysResponse,
+  AppConfigurationGetKeyValuesOptionalParams,
+  AppConfigurationGetKeyValuesResponse,
+  AppConfigurationCheckKeyValuesOptionalParams,
+  AppConfigurationCheckKeyValuesResponse,
+  AppConfigurationGetKeyValueOptionalParams,
+  AppConfigurationGetKeyValueResponse,
+  AppConfigurationPutKeyValueOptionalParams,
+  AppConfigurationPutKeyValueResponse,
+  AppConfigurationDeleteKeyValueOptionalParams,
+  AppConfigurationDeleteKeyValueResponse,
+  AppConfigurationCheckKeyValueOptionalParams,
+  AppConfigurationCheckKeyValueResponse,
+  AppConfigurationGetLabelsOptionalParams,
+  AppConfigurationGetLabelsResponse,
+  AppConfigurationCheckLabelsOptionalParams,
+  AppConfigurationCheckLabelsResponse,
+  AppConfigurationPutLockOptionalParams,
+  AppConfigurationPutLockResponse,
+  AppConfigurationDeleteLockOptionalParams,
+  AppConfigurationDeleteLockResponse,
+  AppConfigurationGetRevisionsOptionalParams,
+  AppConfigurationGetRevisionsResponse,
+  AppConfigurationCheckRevisionsOptionalParams,
+  AppConfigurationCheckRevisionsResponse,
+  AppConfigurationGetKeysNextOptionalParams,
+  AppConfigurationGetKeysNextResponse,
+  AppConfigurationGetKeyValuesNextOptionalParams,
+  AppConfigurationGetKeyValuesNextResponse,
+  AppConfigurationGetLabelsNextOptionalParams,
+  AppConfigurationGetLabelsNextResponse,
+  AppConfigurationGetRevisionsNextOptionalParams,
+  AppConfigurationGetRevisionsNextResponse
+} from "./models";
 
-class AppConfiguration extends AppConfigurationContext {
+export class AppConfiguration extends AppConfigurationContext {
```

Comment:
So this'll make it so it shows up in documentation, right? I think we need to do the whole JSDoc \@hidden, \@internal and \@ignore if we need this. People external to this package should not be calling into here at all.

Reply:
@richardpark-msft Actually, it will not be exposed outside of the SDK. Unless you export it in the src/index.ts. In src/index.ts file we export only the following:


```
export * from "./models";
export { AppConfigurationClientOptions, AppConfigurationClient } from "./appConfigurationClient";
```

So, this generated class will not be exported.

Reply:
Also, this class is exported even in the code in master branch:https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/appconfiguration/app-configuration/src/generated/src/appConfiguration.ts#L759

This export basically enables the custom layer to access this class

Reply:
You're 100% right. It might be that we ignore generated code when generating docs (I'll check!).

Reply:
Ah, okay, this is a bit more interesting than I intended.

So I just ran `npm run docs` which exports the docs as typedoc would see them. When I do that there is a difference in how these two types of exporting are treated:

Export using the `export` keyword directly on the definition of the class causes it to be documented:

```typescript
export class Blah {}
```
 
vs this way (exporting the symbol, but not on the class itself), which does **NOT** cause it to be documented:

```typescript 
class Blah{}
 
export {
  Blah
}
```

I don't think I did this change manually but it's possible I did. Do we know how we normally ignore auto-generated code? I'm not sure if this is just a weird loophole we fell into or if this is the way it's supposed to be done.

@deyaaeldeen have you run into this interesting set of circumstances? I feel like you'll probably know the right way to exclude these.

Reply:
That is interesting. But, our review.md files don't include them.

Reply:
Regarding the docs, `typedoc`, our docs generation tool, does not care if a symbol is exported by an entry point. It documents all exported definitions from all source files if they're not annotated by neither `@hidden` nor `@ignore`. I am not sure if this answers your question though.

Reply:
Thanks @deyaaeldeen , I think it does. What we're doing might just be accidentally taking advantage of a bug in typedoc. The end result is correct though - we don't want to just auto-document all the internals for the generated code (at least, not by default).

@sarangan12, it seems like we need to do something here to avoid that problem. Is it possible to auto-generate the required hidden or ignore attributes?

Reply:
Sure. It is possible to auto-generate it or change the format like the one we have in master branch. Let me first do the change in autorest.typescript and get back to you.

Reply:
Updated autorest.typescript and regenerated the appconfiguration SDK to make sure that client and client context classes (inside the generated layer) are hidden. Checked the output of ```npm run docs``` and confirmed the classes are not showing up. 

Thanks @richardpark-msft for great catch.